### PR TITLE
MOBILE-7016: [iOS][Swift] Migrate Native Assets

### PR DESCRIPTION
**Pull request by VadimKhohlovOpenX**
**JIRA issue**

https://openxtechinc.atlassian.net/browse/MOBILE-7016

**Description of change**

- rewritten PBMNativeAssetTitle on Swift
- rewritten PBMNativeAssetData on Swift
- rewritten PBMNativeAssetImage on Swift
- rewritten PBMNativeAssetVideo on Swift
- rewritten PBMNativeAsset on Swift

PrebidExamplesUITest swift 2021-05-12 09-34-45

### CODE REVIEW

File: PrebidMobile/Rendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/ViewControllers/Configuration/NativeAdConfigEditor/EditorUtils.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//
+//  EditorUtils.swift
+//  PrebidMobileDemoRendering
+//
+//  Copyright © 2021 Prebid. All rights reserved.
+//
+
+import Foundation
+
+class EditorUtils{
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/Rendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/ViewControllers/Configuration/NativeAdConfigEditor/EditorUtils.swift]: Can we make this functionality as an extension for NativeAsset?

File: PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    /// This object is a placeholder that may contain custom JSON agreed to by the parties
+    /// to support flexibility beyond the standard defined in this specification
+    @objc private(set) public var assetExt: [String : Any]?
+    
+    @objc private(set) public var childExt: [String : Any]?
+    
+    var childType: String
+    
+    private override init() {
+        NSLog("NativeAsset class should not be instantialted directly, instantiate subclasses instead.")
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift]: Need to remove NSLog and use internal logging system

File: PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    
+    // MARK: - Protected
+    
+    @objc public func setChildExt(_ childExt: [String : Any]?) throws {
+        let nsExt = childExt as NSDictionary?
+        let newExt = try nsExt?.unserializedCopy()
+        self.childExt = newExt
+    }
+    
+    func appendAssetProperties(_ jsonDictionary: MutableJsonDictionary) {
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift]: need to rename this method in order to  show that it adds properties to  the input parameter, f.e.

appendAssetProperties(to jsonDictionary: MutableJsonDictionary)

File: PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAssetData.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        }
+    }
+    
+    @objc public func setDataExt(_ dataExt: [String : Any]?) throws {
+        try setChildExt(dataExt)
+    }
+
+    // MARK: - Protected
+
+    public override func appendChildProperties(_ jsonDictionary: MutableJsonDictionary) {
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAssetData.swift]: appendChildProperties(to jsonDictionary: MutableJsonDictionary)

File: PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAssetImage.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    
+    // MARK: - Image Ext
+    
+    @objc public func setImageExt(_ imageExt: [String : Any]?) throws {
+        try setChildExt(imageExt)
+    }
+    
+    // MARK: - Protected
+
+    public override func appendChildProperties(_ jsonDictionary: MutableJsonDictionary) {
```
Comment (YuriyVelichkoOpenX) [inline PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAssetImage.swift]: appendChildProperties(to jsonDictionary: MutableJsonDictionary)

**Review by YuriyVelichkoOpenX — CHANGES_REQUESTED** [review]
Please, make small fixes

File: PrebidMobile/Rendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/ViewControllers/Configuration/NativeAdConfigEditor/EditorUtils.swift (hunk 1 above)
Comment (VadimKhohlovOpenX) [reply PrebidMobile/Rendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/PrebidMobileDemoRendering/ViewControllers/Configuration/NativeAdConfigEditor/EditorUtils.swift]: Done

File: PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift (hunk 2 above)
Comment (VadimKhohlovOpenX) [reply PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift]: Done

File: PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift (hunk 3 above)
Comment (VadimKhohlovOpenX) [reply PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCore/NativeAssets/NativeAsset.swift]: Done

**Review by YuriyVelichkoOpenX — APPROVED** [review]
LGTM

**Pull request merged**